ansible/roles/zookeeper/templates/zookeeper.cfg.j2

```diff
@@ -4,5 +4,5 @@ syncLimit=5
 dataDir=/var/lib/zookeeper/data
 clientPort={{ zookeeper_client_port }}
 {% for host in groups['zookeeper'] -%}
-server.{{ loop.index }}={{ hostvars[host]['api_interface_address'] }}:{{ zookeeper_peer_port }}:{{ zookeeper_quorum_port }}
+server.{{ loop.index }}={{ hostvars[inventory_hostname]['ansible_' + api_interface]['ipv4']['address'] }}:{{ zookeeper_peer_port }}:{{ zookeeper_quorum_port }}
```

Comment:
Don't we want the address of `host` here, rather than the current host which is given by `inventory_hostname`? If we look up `api_interface_address` for `host`, it should evaluate `inventory_hostname` for that host, but if we do it here we look up the current host each time. Perhaps I'm missing something.

Reply:
Sorry for the confusion - you're right. I meant to revert it to the original syntax I had here.

Reply:
What was wrong with using api_interface_address?

Reply:
Nothing afaik - I just hadn't tested it so I reverted to the original syntax I'd tested it to save time. I'll check it works. I guess this is what you're after:

```
server.{{ loop.index }}={{ hostvars[host]['ansible_' + api_interface]['ipv4']['address'] }}:{{ zookeeper_peer_port }}:{{ zookeeper_quorum_port }}
```

Reply:
Looks good now with api_interface_address:
```
[root@sv zookeeper]# cat zookeeper.cfg 
tickTime = 2000
initLimit = 10
syncLimit = 5
dataDir = /var/lib/zookeeper/data
clientPort = 2181
server.1 = 192.168.7.3:2888:3888
```

Reply:
I don't mind that much, as long as it works. I suggested using api_interface_address to keep the template short, I thought it was equivalent to what you had originally:

`hostvars[host]['api_interface_address']`

What you have in the last comment isn't quite right, as it will use api_interface of the current host, not the loop host.

Reply:
If I use: ```server.{{ loop.index }}={{ hostvars[host]['api_interface_address'] }}:{{ zookeeper_peer_port }}:{{ zookeeper_quorum_port }}``` it templates to something like server.1=hostvars[inventory_hostname]['ansible_' + api_interface]['ipv4']['address']:2888:3888

Reply:
That's odd, I'd have thought it would recursively evaluate the jinja templates.

Reply:
I checked it again with ```server.{{ loop.index }}={{ hostvars[host]['api_interface_address'] }}:{{ zookeeper_peer_port }}:{{ zookeeper_quorum_port }}``` and it doesn't. Strange:

```
[root@sv-b16-u22 zookeeper]# cat zookeeper.cfg 
tickTime = 2000
initLimit = 10
syncLimit = 5
dataDir = /var/lib/zookeeper/data
clientPort = 2181
server.1 = {{ hostvars[inventory_hostname]['ansible_' + api_interface]['ipv4']['address'] }}:2888:3888

```

Reply:
Did  quick grep in kolla-ansible and can't see api_interface_address being used with hostvars. I wonder if it's a limitation of merge_configs, or perhaps my understanding isn't quite right. Anyway, let's stick with the original and move on :)

Reply:
Yeah, i couldn't see it either. Seemed like a reasonable suggestion though. Thanks for the review!